packages/legend-application-data-cube/src/stores/LegendDataCubeDuckDBEngine.ts

```diff
@@ -251,15 +249,25 @@ export class LegendDataCubeDuckDBEngine {
 
     const data = result.toArray();
     const columnNames = result.schema.fields.map((field) => field.name);
+    const columnTypesIds = result.schema.fields.map((field) => field.typeId);
     const rows = data.map((row) => {
       const tdsRow = new TDSRow();
-      tdsRow.values = columnNames.map((column) => {
+      tdsRow.values = columnNames.map((column, idx) => {
         const value = row[column] as unknown;
         // NOTE: DuckDB WASM returns ArrayBuffer for numeric value, such as for count(*)
         // so we need to convert it to number
         if (ArrayBuffer.isView(value)) {
           return row[column].valueOf() as number;
           // BigInt is not supported by ag-grid, so we need to convert it to native number
```

Comment:
this comment is meant for the block below, you should add a comment to describe why you're doing the conversion here as well for date/time fields